crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,156 @@
+use super::{endpoint::Endpoint, octez_node::OctezNode};
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
+
+#[derive(Default)]
+pub struct OctezClientBuilder {
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    // if None, use localhost:8732 (the default endpoint for octez-node)
+    endpoint: Option<OctezNodeEndpoint>,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientBuilder {
+    pub fn new() -> Self {
+        OctezClientBuilder::default()
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
+        self.binary_path = Some(path);
+        self
+    }
+    pub fn set_base_dir(mut self, base_dir: PathBuf) -> Self {
+        self.base_dir = Some(base_dir);
+        self
+    }
+    pub fn set_endpoint(mut self, endpoint: OctezNodeEndpoint) -> Self {
+        self.endpoint = Some(endpoint);
+        self
+    }
+    pub fn set_disable_unsafe_disclaimer(
+        mut self,
+        disable_unsafe_disclaimer: bool,
+    ) -> Self {
+        self.disable_unsafe_disclaimer = disable_unsafe_disclaimer;
+        self
+    }
+    pub fn build(mut self) -> OctezClient {
+        let base_dir = self
+            .base_dir
+            .map_or(OctezClientDirectory::default(), OctezClientDirectory::Path);
```

Comment:
`OctezClientDirectory::default()` is misleading because the default directory of octez-client is `~/.tezos-client`. Could you instead inline the temp dir creation?